### PR TITLE
Added notification listener

### PR DESCRIPTION
**Pull request by kishlay1826**
On creation of MainActivity, calling a function that listens for addition of new data to 'challenges' table and shows notifications subsequently.

### CODE REVIEW

File: app/src/main/java/com/github/geohunt/app/ui/screens/main/MainViewModel.kt [hunk 1]
```diff
@@ -41,6 +50,32 @@ class MainViewModel(
         }
     }
 
+    fun listenForNewNotification(context: Context) {
+        val database = FirebaseDatabase.getInstance()
```
Comment (Maeeen) [inline app/src/main/java/com/github/geohunt/app/ui/screens/main/MainViewModel.kt:54]: Shouldn't this be put in a repository?

File: app/src/main/java/com/github/geohunt/app/ui/screens/main/MainViewModel.kt [hunk 2]
```diff
@@ -41,6 +50,32 @@ class MainViewModel(
...
+        val challengesRef = database.getReference("challenges")
+
+        val challengesListener = object : ValueEventListener {
+            override fun onDataChange(snapshot: DataSnapshot) {
+                snapshot.children.forEach { child ->
+                    if (child.key != null) {
+                        showNotification(
+                            context = context,
+                            title = "New Challenge!",
```
Comment (Maeeen) [inline app/src/main/java/com/github/geohunt/app/ui/screens/main/MainViewModel.kt:63]: Maybe use `stringResource` for i18n?

File: app/src/main/java/com/github/geohunt/app/ui/screens/main/MainViewModel.kt (hunk 1 above)
Comment (Maeeen) [reply app/src/main/java/com/github/geohunt/app/ui/screens/main/MainViewModel.kt:54]: I mean, it's fine but this should be tested as well, no?

Comment (kishlay1826) [reply app/src/main/java/com/github/geohunt/app/ui/screens/main/MainViewModel.kt:54]: Apparently testing whether notifications show up is very hard and outside the scope of unit test. I searched ways and I'll have to install other libraries and incorporate end to end tests but it's kinda hard

File: app/src/main/java/com/github/geohunt/app/ui/screens/main/MainViewModel.kt (hunk 2 above)
Comment (kishlay1826) [reply app/src/main/java/com/github/geohunt/app/ui/screens/main/MainViewModel.kt:63]: I'm not sure I understand what you mean. We don't need any formatting here, it's just passing title and message for notification. Do you mean a constants file of sort and define the title and message there and then use it here?

Comment (Maeeen) [reply app/src/main/java/com/github/geohunt/app/ui/screens/main/MainViewModel.kt:63]: Basically, every text should be put in the `strings.xml` file to make internationalization a thing.

**Comment by Maeeen** [comment]
LGTM! 👍

**Review by Maeeen — APPROVED** [review]

**Pull request merged**